presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -135,15 +130,10 @@ private void unspillNextLookupSource()
         checkState(spilledPartitions.isPresent());
         if (spilledPartitions.get().hasNext()) {
             int currentSpilledPartition = spilledPartitions.get().next();
```

Comment:
previously there was no need for a contract, that `spilledPartitions` iterator returns always elements in the same order, for each LJO. It should work fine now, but It might be better to refactor this somehow? For example obtain `currentSpilledPartition` value from `lookupSourceFactory` and let it take care of the order of unspilling?